src/Ema/Route/Class.hs

```diff
@@ -1,25 +1,32 @@
 module Ema.Route.Class (
-  IsRoute (RouteModel, routeEncoder, allRoutes),
+  IsRoute (RouteModel, routePrism, routeUniverse),
 ) where
 
-import Ema.Route.Encoder (RouteEncoder, singletonRouteEncoder)
+import Ema.Route.Prism (Prism_, toPrism_)
+import Optics.Core (only)
 import Prelude hiding (All, Generic)
 
 {- | Class of Ema routes
 
   An Ema route has an encoder, that knows how to convert it to/from filepaths.
-  As well as an universe function, `allRoutes`, that gives all possible route
+  As well as an universe function, `routeUniverse`, that gives all possible route
   values in a static site.
 
   Both the encoder and the universe function take the associated model as an argument.
 -}
 class IsRoute r where
   type RouteModel r :: Type
-  routeEncoder :: RouteEncoder (RouteModel r) r
-  allRoutes :: RouteModel r -> [r]
+
+  -- | An optics `Prism'` that denotes how to encode and decode a route.
+  routePrism :: RouteModel r -> Prism_ FilePath r
```

Comment:
Really wish we could use `Prism'` (optics-core) here ... but: https://stackoverflow.com/q/71489589/55246